src/main/java/de/fraunhofer/aisec/cpg/graph/TypeManager.java

```diff
@@ -62,6 +62,9 @@
 
   @NonNull private Map<String, RecordDeclaration> typeToRecord = new HashMap<>();
 
+  @NonNull
```

Comment:
I would add a descriptive comment here. For me comming from the PR it is obvious what type parameters do, but for someone that goes through the `TypeManager` it may be not.

Reply:
The rest looks fine to me, so we can merge later